.circleci/config.yml

```diff
@@ -8,146 +8,166 @@ jobs:
       - checkout
       - restore_cache:
           keys:
-            - cargo-v2-{{ checksum "Cargo.toml" }}-
-            - cargo-v2-
-      - run: cargo update
+            - cargo-v3-{{ checksum "Cargo.toml" }}
+            - cargo-v3-
+      - run: git submodule sync
+      - run: git submodule update --init
       - run: cargo fetch
       - persist_to_workspace:
-          root: "."
+          root: /mnt/crate
           paths:
             - Cargo.lock
       - save_cache:
-          key: cargo-v2-{{ checksum "Cargo.toml" }}-{{ checksum "Cargo.lock" }}
+          key: cargo-v3-{{ checksum "Cargo.toml" }}
           paths:
             - /usr/local/cargo/registry
             - /usr/local/cargo/git
-  test:
+
+  rustfmt:
+    docker:
+      - image: rust:latest
+    working_directory: /mnt/crate
+    steps:
+      - checkout
+      - run:
+          name: Install rustfmt
+          command: rustup component add rustfmt
+      - run:
+          name: Print version information
+          command: cargo fmt -- --version
+      - run:
+          name: Check rustfmt
+          command: cargo fmt -- --check
+
+  build_debug:
     docker:
       - image: rust:latest
     working_directory: /mnt/crate
     steps:
       - checkout
       - attach_workspace:
-          at: "."
+          at: /mnt/crate
+      - run: git submodule sync
+      - run: git submodule update --init
       - restore_cache:
           keys:
-            - cargo-v2-{{ checksum "Cargo.toml" }}-{{ checksum "Cargo.lock" }}
+            - cargo-v3-{{ checksum "Cargo.toml" }}-{{ checksum "Cargo.lock" }}
```

Comment:
Now this line never matches because the key never includes the `Cargo.lock` checksum?